circuit-prover/examples/poseidon_perm_merkle.rs

```diff
@@ -122,15 +122,15 @@ fn main() -> Result<(), Box<dyn Error>> {
     let row0_state_base = flatten_ext_limbs(&row0_state);
     let row0_out_base = perm.permute(row0_state_base);
 
-    // Row 1 chaining: mmcs_bit = 1, so previous hash becomes right child (limbs 0-1 get prev_out[2..4])
-    // limbs 2-3 from sibling1
+    // Row 1 chaining: mmcs_bit = 1, so previous hash becomes right child.
+    // Previous digest (out[0..1]) chains into limbs 2-3; sibling1 provides limbs 0-1.
     let mut row1_state_base = [Base::ZERO; WIDTH];
-    // limbs 0-1 from row0 output limbs 2-3
-    row1_state_base[0..2 * LIMB_SIZE].copy_from_slice(&row0_out_base[2 * LIMB_SIZE..4 * LIMB_SIZE]);
-    // limbs 2-3 from sibling1
+    // limbs 0-1 from sibling1
     let sibling1_flat =
         flatten_ext_limbs(&[sibling1_limb2, sibling1_limb3, Ext4::ZERO, Ext4::ZERO]);
```

Comment:
You can remove the extra zeroes if you make `flatten_ext_limbs` generic over the input and output sizes.